app/Http/Controllers/Table/ProcessorsController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers\Table;
+
+use App\Models\Processor;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Blade;
+use LibreNMS\Util\Html;
+use LibreNMS\Util\Url;
+
+class ProcessorsController extends TableController
+{
+    protected $default_sort = ['device_hostname' => 'asc', 'processor_descr' => 'asc'];
+
+    protected function sortFields($request): array
+    {
+        return [
+            'device_hostname',
+            'processor_descr',
+            'processor_usage',
+        ];
+    }
+
+    protected function searchFields(Request $request): array
+    {
+        return [
+            'device_hostname',
+            'processor_descr',
+        ];
+    }
+
+    protected function baseQuery(Request $request): Builder
+    {
+        return Processor::query()
+            ->hasAccess($request->user())
+            ->with(['device', 'device.location'])
+            ->withAggregate('device', 'hostname');
+    }
+
+    /**
+     * @param  Processor  $processor
+     */
+    public function formatItem($processor): array
+    {
+        $perc = round($processor->processor_usage);
+        $graph_array = [
+            'type' => 'processor_usage',
+            'popup_title' => htmlentities(strip_tags($processor->device->displayName() . ': ' . $processor->processor_descr)),
+            'id' => $processor->processor_id,
+            'from' => '-1d',
+            'height' => 20,
+            'width' => 80,
+        ];
+
+        $hostname = Blade::render('<x-device-link :device="$device" />', ['device' => $processor->device]);
```

Comment:
Not the biggest fan of this, I guess its fine tho

Reply:
Open to real solutions :)